Neos.Flow/Classes/ObjectManagement/Configuration/ConfigurationBuilder.php

```diff
@@ -561,7 +561,13 @@ protected function autowireProperties(array &$objectConfigurations)
                 if (!array_key_exists($propertyName, $properties)) {
                     /** @var Inject $injectAnnotation */
                     $injectAnnotation = $this->reflectionService->getPropertyAnnotation($className, $propertyName, Inject::class);
-                    $objectName = $injectAnnotation->name !== null ? $injectAnnotation->name : trim(implode('', $this->reflectionService->getPropertyTagValues($className, $propertyName, 'var')), ' \\');
+                    $objectName = $injectAnnotation->name;
+                    if ($objectName === null) {
+                        $objectName = $this->reflectionService->getPropertyType($className, $propertyName);
+                    }
+                    if ($objectName === null) {
+                        $objectName = trim(implode('', $this->reflectionService->getPropertyTagValues($className, $propertyName, 'var')), ' \\');
+                    }
```

Comment:
would that work or is this to fancy 😁

```suggestion
                    $objectName = $injectAnnotation->name
                        ?? $this->reflectionService->getPropertyType($className, $propertyName)
                        ?? trim(implode('', $this->reflectionService->getPropertyTagValues($className, $propertyName, 'var')), ' \\');
```

or maybe this `??=`?

```suggestion
                    $objectName = $injectAnnotation->name
                    $objectName ??= $this->reflectionService->getPropertyType($className, $propertyName)
                    $objectName ??= trim(implode('', $this->reflectionService->getPropertyTagValues($className, $propertyName, 'var')), ' \\');

```

Reply:
Thanks for the suggestion!

If you wonder if this would work, I'd say it's too fancy.

Reply:
true i have never seen a chained null coalescing - but i works ^^

https://3v4l.org/FtNJV

(and i love fancy 😁)

Reply:
Alright, let's go fancy then 🕺